wolfssl/wolfcrypt/cmac.h

```diff
@@ -74,12 +82,15 @@ struct Cmac {
 
 
 
-typedef enum CmacType {
-    WC_CMAC_AES = 1
-} CmacType;
 
+#ifndef NO_AES
 #define WC_CMAC_TAG_MAX_SZ AES_BLOCK_SIZE
 #define WC_CMAC_TAG_MIN_SZ (AES_BLOCK_SIZE/4)
+#else
+/* Reasonable defaults */
+#define WC_CMAC_TAG_MAX_SZ 16
+#define WC_CMAC_TAG_MIN_SZ 4
+#endif
```

Comment:
Since AES is the only cipher supported by CMAC for now, should we consider adding a compile time error if CMAC is enabled without AES? Or was this deliberate to allow for cryptoCbs to still work?

Reply:
This was deliberate to support other types if the cryptocb wanted to support it.  The functions return FUNC_BAD_ARG if an unknown type is passed, so at least this will be detected at run time